src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -2144,18 +2145,108 @@ void MacroAssembler::check_klass_subtype_slow_path(Register sub_klass,
   bdnz(loop);
 
   bind(failure);
-  if (result_reg!=noreg) li(result_reg, 1); // load non-zero result (indicates a miss)
+  if (result_reg != noreg) {
+    li(result_reg, 1); // load non-zero result (indicates a miss)
+  } else if (L_success == nullptr) {
+    crandc(CCR0, Assembler::equal, CCR0, Assembler::equal); // miss indicated by CR0.ne
+  }
   b(fallthru);
 
   bind(hit);
   std(super_klass, target_offset, sub_klass); // save result to cache
   if (result_reg != noreg) { li(result_reg, 0); } // load zero result (indicates a hit)
   if (L_success != nullptr) { b(*L_success); }
```

Comment:
Handling `L_success != nullptr` should be put on the else-branch of the previous if-statement.

Reply:
Doesn't make a real difference, but I've cleaned it up.

Reply:
Thanks. It matches the assertion you've added. I like consistency. It helps understanding stuff.